market_maker_keeper/dydx_market_maker_keeper.py

```diff
@@ -102,24 +102,28 @@ def our_available_balance(self, our_balances: dict, token: str) -> Wad:
         if token == 'weth':
             token = 'eth'
 
-        wei_balance = list(filter(lambda x: x['currency'] == token.upper(), our_balances))[0]['wei']
-        balance = from_wei(abs(int(float(wei_balance))), 'ether')
+        wei_balance = float(list(filter(lambda x: x['currency'] == token.upper(), our_balances))[0]['wei'])
 
         ## DyDx can have negative balances from native margin trading
+        is_negative = False
+        if wei_balance < 0:
+           is_negative = True
+
+        balance = from_wei(abs(int(wei_balance)), 'ether')
+
         # reconvert Wad to negative value if balance is negative
```

Comment:
Does `Wad` or `from_wei` not support negative numbers?

Reply:
yep, `from_wei` does not support negative numbers